File: LeetcodeEasyProblems/_268. Missing Number/Solution.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Use the exclusive OR bitwise operations
        res = len(nums)
        for i in range(len(nums)):
            res ^= i ^ nums[i]
        return res

                

File: LeetcodeEasyProblems/_268. Missing Number/Solution.java
import java.util.HashSet;
class Solution {
    public int missingNumber(int[] nums) {
        HashSet<Integer> numsSet = new HashSet<Integer>();
        for (int num : nums){
            numsSet.add(num);
        }

        for (int i = 0; i < nums.length+1;i++){
            if (!numsSet.contains(i)){
                return i;
            }
        }
        return -1;
    }
}